server/vertx-server/cats/src/main/scala/sttp/tapir/server/vertx/cats/streams/fs2.scala

```diff
@@ -173,36 +176,78 @@ object fs2 {
           }
         }
 
-      override def webSocketPipe[REQ, RESP](
-          readStream: ReadStream[WebSocketFrame],
+      private def decodeFrame[REQ, RESP](
+          frame: WebSocketFrame,
+          o: WebSocketBodyOutput[streams.Pipe[REQ, RESP], REQ, RESP, _, Fs2Streams[F]]
+      ): REQ = {
+        o.requests.decode(frame) match {
+          case DecodeResult.Value(v)         => v
+          case failure: DecodeResult.Failure => throw new WebSocketFrameDecodeFailure(frame, failure)
+        }
+      }
+
+      private def fastPath[REQ, RESP](
+          in: Stream[F, WebSocketFrame],
           pipe: streams.Pipe[REQ, RESP],
           o: WebSocketBodyOutput[streams.Pipe[REQ, RESP], REQ, RESP, _, Fs2Streams[F]]
       ): ReadStream[WebSocketFrame] = {
-        val stream0 = fromReadStreamInternal(readStream)
-        val stream1 = optionallyContatenateFrames(stream0, o.concatenateFragmentedFrames)
-        val stream2 = optionallyIgnorePong(stream1, o.ignorePong)
-        val autoPings = o.autoPing match {
-          case Some((interval, frame)) =>
-            Stream.awakeEvery(interval).as(frame)
-          case None =>
-            Stream.empty
-        }
+        val contatenatedFrames = optionallyContatenateFrames(in, o.concatenateFragmentedFrames)
+        val ignorePings = optionallyIgnorePong(contatenatedFrames, o.ignorePong)
 
-        val stream3 = stream2
-          .map { frame =>
-            o.requests.decode(frame) match {
-              case DecodeResult.Value(v) =>
-                v
-              case failure: DecodeResult.Failure =>
-                throw new WebSocketFrameDecodeFailure(frame, failure)
-            }
-          }
+        val stream = ignorePings
+          .map(decodeFrame(_, o))
           .through(pipe)
           .map(o.responses.encode)
-          .mergeHaltL(autoPings)
+          .chunks
```

Comment:
**question:** Do we need to chunk and then call `Stream.chunk` right after it? This results in the same stream. I think we won't get any benefits by chunking at this point and these two lines can be removed.